pytorch_lightning/strategies/strategy.py

```diff
@@ -100,6 +106,9 @@ def connect(self, model: Module) -> None:
         """Called by the accelerator to connect the accelerator and the model with this plugin."""
         self.model = model
 
+    def _configure_launcher(self):
+        """Attach the launcher based on Strategy."""
+
```

Comment:
Why are we adding this to the base Strategy? Shouldn't this be added to ParallelStrategy instead?

Reply:
inside the accelerator_connector, we just make a call for it, no matter what the strategy is, if you tink, I can add a condition there to make a call for it only if it is a ParallelStrategy.

Reply:
IMO yes. Thoughts about adding this just to `ParallelStrategy` @awaelchli?

Reply:
It makes more sense to me to add it to the base class. Every strategy should have the possibility to choose a launcher, there is conceptually no restriction necessary for it to be exclusively on the ParallelStrategy.

But we we should also keep in mind:

The only reason why we have this method (correct me if I'm wrong @rohitgr7) is because we need to create the launcher lazily. Hence, this `_configure_launcher` method will become entirely obsolete with #7650. Until then, I suggest to keep it protected on the strategy base class.

Reply:
yes! exactly @awaelchli 
let's keep this in base for now

Reply:
Fine with me at the moment.

> Every strategy should have the possibility to choose a launcher, there is conceptually no restriction necessary for it to be exclusively on the ParallelStrategy.

While there's no technical restriction, I still think it adds confusion that non-parallel strategies could configure process launchers.

If the `ParallelStrategy` is not the right abstraction for you then perhaps we should think of a middle step like

`Strategy -> LaunchingStrategy -> ParallelStrategy`

(not for this PR, you can resolve).